### ISSUE TITLE
Metrics are not retrieved for autoscaling instances launched after logstash

### ISSUE THREAD

**Issue by jstangroome** [EagerELK/logstash-input-cloudwatch#6]
When this plugin initialises the list of instance IDs from the provided tags, the results are stored and used indefinitely.

This appears to be due to this code:

```
  def instance_ids
    return @instances unless @instances.nil?
```

When new instances are launched (or terminated) with matching tags, this plugin needs to requery the current instance IDs.

**Comment by jrgns** [comment]
Great point. I was thinking of putting a timeout on the fetched instances so that the plugin only checks for new instances every 300 seconds. I'll make the timeout configurable as well.

Will that work for you?

**Comment by jstangroome** [comment]
A configurable timeout for checking for new and removed instances, with a default of 300 seconds seems like a good solution.

**Comment by jrgns** [comment]
I've created a prelim fix for this. I need to test it properly, though.

If you're brave, feel free to test this somewhere and report back! :)

**Comment by jrgns** [comment]
Released this as part of v0.2.0
